homeworks/05-redux-basic/002-basic-redux/src/components/BalanceControl.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react"
+import { connect } from "react-redux"
+import {
+  credit,
+  debit,
+  substractPercentage,
+  updateBalance,
+} from "../redux/actions"
+
+interface PropsFromRedux {
+  updateBalance: (amount: number) => void
+  credit: (amount: number) => void
+  debit: (amount: number) => void
+  substractPercentage: (taxAmountPercentage: number) => void
+}
+
+interface BalanceControlProps extends PropsFromRedux {}
+
+const BalanceControlComponent = ({
+  updateBalance,
+  credit,
+  debit,
+  substractPercentage,
+}: BalanceControlProps): React.ReactElement => {
+  const onUpdateBalance = React.useCallback(() => {
+    updateBalance(2000)
+  }, [updateBalance])
+  const onCredit = React.useCallback(() => {
+    credit(350)
+  }, [credit])
+  const onDebit = React.useCallback(() => {
+    debit(500)
+  }, [debit])
+  const onSubstractPercentage = React.useCallback(() => {
+    substractPercentage(13)
+  }, [substractPercentage])
+
+  return (
+    <div>
+      <button onClick={onUpdateBalance}>updateBalance 2000</button>
+      <button onClick={onCredit}>credit 350</button>
+      <button onClick={onDebit}>debit 500</button>
+      <button onClick={onSubstractPercentage}>substractPercentage 13</button>
+    </div>
+  )
+}
+
+const mapDispatchToProps = (dispatch: Function): PropsFromRedux => ({
```

Comment:
i do think that name is like `PropsFromRedux` is overkilling)

Reply:
Thanks! You're right.

I've pushed fix-commit.
I used `ReturnType<typeof mapDispatchToProps>` instead of `PropsFromRedux`, which seems legit :)